<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PhongLight</class>
 <widget class="QDialog" name="PhongLight">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>879</width>
    <height>529</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="PhongLightBox">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>50</height>
      </size>
     </property>
     <property name="title">
      <string>Phong光照模型</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_10">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="colorUniformBox_2">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>50</height>
           </size>
          </property>
          <property name="title">
           <string>光源的颜色：</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="5,1,3,1,3,1,3">
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>r：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Light_X_SpinBox">
             <property name="minimum">
              <double>0.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>g：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Light_Y_SpinBox">
             <property name="minimum">
              <double>0.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>b：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Light_Z_SpinBox">
             <property name="minimum">
              <double>0.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QGroupBox" name="colorUniformBox_6">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>50</height>
             </size>
            </property>
            <property name="title">
             <string>光源的位置：</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="5,1,3,1,3,1,3">
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QLabel" name="label_21">
               <property name="text">
                <string>x:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="Pos_X_SpinBox">
               <property name="minimum">
                <double>-10000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
               <property name="value">
                <double>1.200000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_22">
               <property name="text">
                <string>y：</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="Pos_Y_SpinBox">
               <property name="minimum">
                <double>-10000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
               <property name="value">
                <double>1.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_23">
               <property name="text">
                <string>z：</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="Pos_Z_SpinBox">
               <property name="minimum">
                <double>-10000.000000000000000</double>
               </property>
               <property name="maximum">
                <double>1000000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>0.100000000000000</double>
               </property>
               <property name="value">
                <double>2.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="colorUniformBox_3">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>50</height>
           </size>
          </property>
          <property name="title">
           <string>物体的颜色：</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="5,1,3,1,3,1,3">
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>r：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Obj_X_SpinBox">
             <property name="minimum">
              <double>0.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>g：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Obj_Y_SpinBox">
             <property name="minimum">
              <double>0.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>0.500000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>b：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="Obj_Z_SpinBox">
             <property name="minimum">
              <double>0.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>0.300000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="colorUniformBox_7">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>50</height>
           </size>
          </property>
          <property name="title">
           <string>环境光、漫反射和镜面反射</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_13">
           <item>
            <spacer name="horizontalSpacer_10">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_24">
             <property name="text">
              <string>ambient强度：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="PhongAmbient_SpinBox">
             <property name="minimum">
              <double>0.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>0.100000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_25">
             <property name="text">
              <string>diffuse：通过Normal信息计算</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_26">
             <property name="text">
              <string>specular强度：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="PhongSpecular_SpinBox">
             <property name="minimum">
              <double>0.000000000000000</double>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>0.500000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_27">
             <property name="text">
              <string>specular-shiny：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="shiny_spinBox">
             <property name="maximum">
              <number>1000000</number>
             </property>
             <property name="value">
              <number>32</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
